pandas/core/dtypes/cast.py

```diff
@@ -1055,7 +1055,7 @@ def maybe_cast_to_datetime(value, dtype, errors="raise"):
                     )
 
             if is_scalar(value):
-                if value == iNaT or isna(value):
+                if value is iNaT or isna(value):
```

Comment:
@jbrockmendel something to address later, we need to separate integer tests here

@inmoonlight is there a reason you changed this?

Reply:
The integer iNaT is not interned, so this seems dangerous

Reply:
@jreback 
I found this led to the failure in `pandas-dev.pandas (Linux py37_np_dev)` env. If you all think it is dangerous, I'll put it back.

Reply:
Restore the original I think. We've had some flaky tests in master recently that we're fixing elsewhere.

Reply:
@TomAugspurger 
Restored it.